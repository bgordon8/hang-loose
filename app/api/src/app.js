import express from 'express'

const app = express()

app.get("/", (req, res) => {
    res.status(200).json({
        message: "hello world!!!! thats whats up!!!"
    })
})

if (process.env.NODE_ENV !== "test") {
    app.listen(3000, () => {
        console.log('listening on port 3000...')
    })
}

export default app