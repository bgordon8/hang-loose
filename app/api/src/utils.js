import bcrypt from 'bcryptjs'
import JWT from 'jsonwebtoken'
import dayjs from 'dayjs'

async function encodeToken(user) {
  const tokenSecret = process.env.TOKEN_SECRET || 'cowabunga'

  return {
    token: JWT.sign(
      {
        exp: dayjs().add(15, 'day').unix(),
        iat: dayjs().unix(),
        sub: user.id,
      },
      tokenSecret,
      {}
    ),
  }
}

async function comparePass(userPassword, dbPassword) {
  const valid = await bcrypt.compareSync(userPassword, dbPassword)

  if (valid) {
    return true
  } else {
    throw new Error('invalid credentials')
  }
}

export { encodeToken, comparePass }
