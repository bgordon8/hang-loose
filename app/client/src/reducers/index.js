import { combineReducers } from 'redux'
import channel from './channel-reducer'
import auth from './auth-reducer'

const rootReducer = combineReducers({
  auth,
  channel,
})

export default rootReducer
