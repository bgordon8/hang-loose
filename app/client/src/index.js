import React from 'react'
import ReactDOM from 'react-dom'
import { Provider } from 'react-redux'
import { createStore, applyMiddleware } from 'redux'
import { BrowserRouter as Router } from 'react-router-dom'
import logger from 'redux-logger'
import reduxThunk from 'redux-thunk'
import reducers from './reducers'
import App from './components/App'
import { AUTH_USER } from './actions/types'

const store = applyMiddleware(logger, reduxThunk)(createStore)(reducers)
const token = localStorage.getItem('token')

if (token) {
  store.dispatch({ type: AUTH_USER })
}

ReactDOM.render(
  <React.StrictMode>
    <Provider store={store}>
      <Router>
        <App />
      </Router>
    </Provider>
  </React.StrictMode>,
  document.getElementById('root')
)
