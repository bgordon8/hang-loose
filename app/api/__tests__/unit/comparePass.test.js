import bcrypt from 'bcryptjs'
import { comparePass } from '../../src/utils'

describe('comparePass', () => {
  it('returns true if the passwords match', async () => {
    const salt = bcrypt.genSaltSync()
    const hash = bcrypt.hashSync('password123', salt)

    const res = await comparePass('password123', hash)

    expect(res).toBe(true)
    expect(res).toMatchSnapshot()
  })
})
