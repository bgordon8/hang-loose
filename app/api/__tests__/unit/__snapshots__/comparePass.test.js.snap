// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`comparePass returns true if the passwords match 1`] = `true`;
